packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -805,7 +805,7 @@ abstract class FlutterCommand extends Command<void> {
   ///
   /// Throws a [ToolExit] if the current set of options is not compatible with
   /// each other.
-  Future<BuildInfo> getBuildInfo({ BuildMode forcedBuildMode }) async {
+  Future<BuildInfo> getBuildInfo({ BuildMode forcedBuildMode, bool updateWebDefines = true }) async {
```

Comment:
Passing along a bool for web mode seems a bit special purpose. Is there something that could be passed here instead that would convey the same thing, but which could also encode other platforms. Maybe `TargetPlatform`?

Reply:
Definitely. though in this case we can have multiple target platforms - I could pass a Set< TargetPlatform> . I think in this case though, we should (in due time) move the update to the web dart defines into the web build targets and resident web runner, since that is very web specific.

Reply:
That would remove the need to handle any target platform logic here

Reply:
If the change to move the logic into the web-specific code is small, that does seem like a better place, but if that change will be higher-risk, then filing an issue and adding a TODO sgtm.